main/network.c

```diff
@@ -1287,7 +1287,7 @@ struct hostent * gethostname_re (const char *host,struct hostent *hostbuf,char *
 	}
 
 	while (( res =
-		gethostbyname_r(host,hostbuf,*tmphstbuf,*hstbuflen,&hp,&herr))
+		gethostbyname2_r(host,AF_INET,hostbuf,*tmphstbuf,*hstbuflen,&hp,&herr))
```

Comment:
Can this be fixed without using gethostbyname2_r? This file already tests many different gethostbyname prototypes for cross-platform support, and we shouldn't add more unless unavoidable.

The current change is incorrect as-is, because it assumes that gethostbyname2_r is available whenever a six-argument prototype of gethostbyname_r is available.

Reply:
I don't think it can be properly fixed without it. As in the original bug report, it was possible for a domain with a valid A record yet no AAAA record return a result for AF_INET6. In this system call we need to explicitly request an AF_INET result only. 

Any result from gethostbyname_r with an INET6 return struct is useless in this context, it doesn't indicate that there is no record for a normal ipv4 address.

Reply:
I this @nikic is right, also the issue would remain on other platforms as Linux is the only one to support `gethostbyname2_r` for specifying a `type`.
I think we can use `getaddrinfo` instead of `gethostbyname_r` or `gethostbyname2_r` (both marked as obsolescent). That works basically the same, but returns another `struct addrinfo` instead of `struct hostent` (shouldn´t be an issue, can be manually assigned) and as far as I could tell, the syntax/number of arguments is always the same (checked Linux, Solaris & HP-UX), so we should be able to remove the platform check, don´t we?